code/code/misc/garble.cc

```diff
@@ -224,6 +224,8 @@ sstring TBeing::garble(TBeing *to, const sstring &arg, Garble::SPEECHTYPE speech
 // The speakers ability to "talk good". Since this is an accent and not a language per se this makes sense.
 int getLanguageChance(const TBeing *from, TBeing *to, spellNumT language)
 {
+  TBeing * tFrom = const_cast<TBeing *>(from); // copying this from another function within garble.cc - need to check skill
```

Comment:
Why const_cast? Sounds really, really dangerous...

Reply:
The short of it is these functions pass in the TBeing the message is from as a const, but since we're going to call functions like bSuccess (which have a chance to hone skills and thus need to be able to write to the TBeing data), we need it not to be a const.  I pulled this const_cast code from elsewhere in the Garble code, as this situation has come up previously and this is how they worked around it (while acknowledging it's not clean).

Now with that said... probably the best way to do this is to refactor the relevant Garble functions not to pass in TBeing *from as a const.  I don't know if that would have other repercussions, but I'm guessing since this is the 2nd time it's being recast, that's probably the way we should go.  I don't mind attempting to tackle that, but please let me know if you see any problems going that route before I do.

Reply:
On deeper inspection, the rabbit hole goes deeper than I want to refactor right now.  To remove the const_casts from garble (and mine is only 1 of several), we would need to refactor not just all of garble but also several core info.cc functions.  I think before making that many changes, we should first consider whether it may be easier to just overhaul the communication framework to follow a more modern design - in which case we'd just be rewriting a lot of it anyways.  As such, I'd like to hold off removing the const_cast for now.